pennylane/gradients/pulse_gradient.py

```diff
@@ -278,12 +320,38 @@ def ansatz(params):
     ...     dev,
     ...     interface="jax",
     ...     diff_method=qml.gradients.stoch_pulse_grad,
-    ...     num_split_times=20, # Use 20 samples for the approximation
+    ...     num_split_times=5, # Use 5 samples for the approximation
     ...     sampler_seed=18, # Fix randomness seed
     ... )
     >>> jax.grad(qnode)(params)
-    [Array(0.37935182, dtype=float32, weak_type=True),
-     Array(-0.26384923, dtype=float32, weak_type=True)] # results do not vary
+    [Array(0.00246266, dtype=float32, weak_type=True),
+     Array(-0.11399216, dtype=float32, weak_type=True)]
+
+    On simulator devices, we may activate the option ``use_broadcasting``, which makes
+    use of broadcasting internally to improve the performance of the stochastic parameter-shift
+    rule:
+
+    >>> from time import process_time
+    >>> faster_grad_qnode = qml.QNode(
+    ...     ansatz,
+    ...     dev,
+    ...     interface="jax",
+    ...     diff_method=qml.gradients.stoch_pulse_grad,
+    ...     num_split_times=5, # Use 20 samples for the approximation
+    ...     sampler_seed=18, # Fix randomness seed
+    ...     use_broadcasting=True, # Activate broadcasting
+    ... )
+    >>> for node in [qnode, faster_grad_qnode]:
+    ...     start = process_time()
+    ...     jax.grad(node)(params)
+    ...     print(process_time() - start) # Show the gradient computation time in seconds.
+    11.582010403000002
+    3.9708045299999988
```

Comment:
[personal preference] I'm not a fan of the console style code snippets and would put it in code blocks (same for example above)